Teacher docs (#73)
dodona-edu/dodona-edu.github.io: en/guides/user-management/README.md

---
title: User management
description: "Tutorial: user management"
---

# User management

> All **users** can [sign in](/en/for-students#sign-in) and manage their [user profile](/en/for-students#user-profile). Some actions on Dodona require more [permissions](#permissions). For this reason, we use three roles: [students](#student), [teachers](#teacher) and [admins](#admin). These roles can be [assigned](#assign-roles) to users, for instance a teacher can appoint another teacher.

Reviewer: The last sentence is incorrect, I think. Only admins can assign user roles.

::: tip

> You can see that you are a [teacher](#teacher) when you can see the **management menu** (menu <span class="guilabel">Admin</span>) left of the [user menu](/en/for-students#user-menu) at the right side of the [navigation drawer](/en/for-students#navigation-drawer).

Reviewer: The management menu is now incorporated into the drawer:
![HkpR](https://user-images.githubusercontent.com/3226995/90104672-f835a000-dd44-11ea-841d-39a31a6f122b.png)

![image](./staff.admin_menu.en.png)
:::

::: tip Important

> We divide the [enrolled](/en/for-students#course-register) members of a [course](/en/course-management#course) in [students](#student) and

Reviewer: ```suggestion
We categorize [enrolled](/en/for-students#course-register) members of a [course](/en/course-management#course) in [students](#student) and
```

> [course managers](/en/course-management#course-manager) depending on the [management permissions](#management-permissions)they have been granted for the course. Only course managers can manage a course. The 

Reviewer: ```suggestion
[course managers](/en/course-management#course-manager) depending on the [management permissions](#management-permissions)they have been granted for the course. Only course managers can manage a course.
```

> [managing](/en/course-management#managing-course-users) of 

Reviewer: ```suggestion
[Managing](/en/course-management#managing-course-users)
```

[course users](/en/course-management#course-user) and [appointing](/en/course-management#appointing-course-managers) course managers is discussed in [course management](/en/course-management).
:::

## Navigating to a user

> As a [teacher](#teacher), you navigate to the [user overviewt](#user-overview) by selecting <span class="guilabel">Users</span> in the [management menu](#management-menu) at the right side of the [navigation panel](/en/for-students#navigation-panel).

Reviewer: ```suggestion
As a [teacher](#teacher), you navigate to the [user overview](#user-overview) by selecting <span class="guilabel">Users</span> in the [management menu](#management-menu) at the right side of the [navigation panel](/en/for-students#navigation-panel).
```

Reviewer: The location of the management menu moved to the drawer.

![admin menu users](./staff.admin_menu_users.en.png)

> ## User overview

Reviewer: The screenshots and documentation in this section are using the course wide users table. I think this section needs a bit of a rewrite to reflect the current situation.

Teachers should now search for users within their course page since the general users table has become admin-only.

In the **user overview**, the user's username, real name, email and [progress statistics](#user-progress-statistics) are listed from the [user profile](/en/for-students#user-profile) from all users. In the first column, you can find an [icon](#user-permissions-icon) that corresponds to the [management permissions](#management-permissions) that have been [assigned](#assign-permissions) to that user.

![image](./staff.users.en.png)

## Searching a user
Use the search bar at the top of the [user overview](#user-overview) to search for specific users based on their usernames, names or email addresses.

![image](./staff.users_filtered.en.png)

## Selecting a user

> Click on the username to [navigate](#navigating-to-a-user) to the [profile page](/en/for-students#profile-page) of te user.

Reviewer: ```suggestion
Click on the username to [navigate](#navigating-to-a-user) to the [profile page](/en/for-students#profile-page) of the user.
```

![image](./staff.users_filtered_link.en.png)

## User progress statistics voortgangsstatistieken
The **progress statistics** of a user consists of two numbers $c/s$, where $s$ is the number of [exercises](/en/for-students#exercise) for which that user has [submitted](/en/for-students#submitting-a-solution) atleast one [solution](/en/for-students#solution) and $c$ is the number of [exercises](/en/for-students#exercise) for which that user has [submitted](/en/for-students#submitting-a-solution) atleast one *correct* [solution](/en/for-students#solution).

> ## Editing a user profile

Reviewer: This is not possible anymore for teachers, so I think this section can be removed.

As a [teacher](#teacher), you can edit the [user profile](/en/for-students#user-profile) of [students](#student) and other teachers. To do this, click the edit button in the top right corner of the [profile page](/en/for-students#profile-page) of the user.

![image](./staff.user_edit_link.en.png)

Alternatively, you can press the edit button at the right side of a user in the [user overview](#user-overview) to immediately edit their [user profile](/en/for-students#user-profile)

![image](./staff.users_filtered_edit_link.en.png)

## Management permissions

In order to perform certain action on Dodona, you require additional **management permissions**. As a [teacher](#teacher), you can use the property <span class="guilabel">Permissions</span> on the [profile page](/en/for-students#profile-page) of a user to find out which permissions that user has.

![image](./staff.user_edit_permission.en.png)

> ## User permissions icon

Reviewer: The icons have changed too. There are now two options: course admins (with icon) and students (without icon).
![tYKp](https://user-images.githubusercontent.com/3226995/90105855-dfc68500-dd46-11ea-877c-9d4f60a55cfa.png)

Author: ``` ruby
          <% if user.zeus? %>
            <i class="mdi mdi-android mdi-18" title='<%= t ".zeus" %>'></i>
          <% end %>,
```
 een cursusbeheerder gaat nooit dat icoontje zien?

Reviewer: No, the code responsible for the icons is here:
https://github.com/dodona-edu/dodona/blob/e25509f4862949d01f10c45e1b51ba896c326e64/app/views/course_members/_members_table.html.erb#L19-L24

Only course admins and users not connected to an institution will have an icon. But the latter doesn't need to be documented.

 |<span class="guilabel">Zeus</span>|      ![image](../../../images/role_icons/zeus.png)|    [admins](#admin) have all the permissions [teachers](#teacher) and course administrators have and can additionally assign all [management permissions](#management-permissions) (including course managers), [edit](/en/for-students#edit-user-profile) the [profile](/en/for-students#user-profile) of all users, [edit](/en/course-management#edit-course) all [courses](/en/course-management#course), [edit](/en/course-management#edit-exercise) all [exercises](/en/for-students#exercise) and [edit]((/en/judges#edit-judge)) all [judges](/en/for-students#judge).

### Student
A user with <span class="guilabel">Student</span> [management permissions](#management-permissions)permissions is called a **student**.
### Teacher
A user with <span class="guilabel">Staff</span> [management permissions](#management-permissions)permissions is called a **teacher**.
### Admin
A user with <span class="guilabel">Zeus</span> [management permissions](#management-permissions)permissions is called an **admin**.

> ## Assigning permissions

Reviewer: Assigning site-wide permissions is also not possible anymore as a teacher.

They can however make enrolled members in their coruses _course admin_. That's done trough the 'make course admin' button in the users table of a course.

When you [edit](/en/for-students#user-profile-edit) the [profile](/en/for-students#gebruikersprofiel) of a user,
you can also assign [permissions](#management-permissions) to that user. A teacher can only assign the <span class="guilabel">Student</span> and
<span class="guilabel">Staff</span> permissions to other users (including themselves). In this fashion you can promote a user to a teacher or demote them to a [student](#student).
:::

![image](./staff.user_edit_permission.en.png)

::: tip Important

If you are a [teacher](#teacher) and you demote yourself to [student](#student), you cannot promote yourself back to teacher.
:::

> ## Impersonating users

Reviewer: This is also not possible anymore as a teacher.
